1-JavaScript/Week3/homework/ex5-shoppingCartRedux.js

```diff
@@ -12,8 +12,18 @@ it pure. Do the following:
 4. When constructing the new shopping cart array you should make use of the ES5 
    spread syntax.
 ------------------------------------------------------------------------------*/
-function addToShoppingCart(/* TODO parameter(s) go here */) {
+function addToShoppingCart(arr, item) {
   // TODO complete this function
+  let newArr = [];
+  if(arr.length <= 2){
+      newArr = [...arr];
+      newArr.push(item)
+   } else {
+      newArr = [...arr];
+      newArr.push(item);
+      newArr.shift()
+       }
+       return newArr
```

Comment:
Hi Rana - you could improve this slightly like below:

```
function addToShoppingCart(arr, item) {
  const newArr = [ ...arr ];

  if (newArr.length <= 2) {
    newArr.push(item);
  } else {
    newArr.push(item);
    newArr.shift();
  }
}
```

There are also a couple missing semicolons here. Please take care of it to be consistent 🙂

Reply:
You could also do one better actually. Something like this:

```
function addToShoppingCart(arr, item) {
  const newArr = [ ...arr ];
  newArr.push(item);
  if (newArr.length > 3) {
    newArr.shift();
  }
}
```